### ISSUE TITLE
Inconsistent access to number type keys in Map

### ISSUE THREAD

**Issue by schovi** [facebook/immutable-js#593]
Hi,

I found one inconsistency in Immutable.Map class when using numbers as keys.

    Immutable.Map({ 1: "a" }).get(1) // => undefined
    Immutable.fromJS({ 1: "a" }).get(1) // => undefined
    Immutable.Map().set(1, "a").get(1) // => "a"

When getting JS object from that all ecamples returns `Object { 1: "a" }`
Is there any special behaviour for that or it is just regular bug?

**Comment by Dashed** [comment]
I think for the plain object, `{ 1: "a" }`, the key `1` is converted to the string `'1'`. All keys for plain objects are interpreted as strings.

**Comment by schovi** [comment]
Then last case with `set` is broken or it should works like that?

This really makes me lot of problems with debugging :)

**Comment by Dashed** [comment]
This works as expected:

```js
Immutable.Map().set(1, "a").get(1) // => "a"
```

-----

See:

```js
const Immutable = require('immutable');
const foo = Immutable.Map().set(1, "a").set('1', "b");
console.log(foo + ''); // => Map { 1: "a", "1": "b" }
```

-----

`Immutable.Map` behaves like a proper map/dictionary. If you're using plain object convention, you'll probably want to marshal your keys with `String()`: 

```js
const keyify = String;
const foo = Immutable.Map().set(keyify(1), "a");
```

**Comment by samwgoldman** [comment]
This also came up in #588 where you can find a response from @leebyron. I have an open PR to document this behavior in #589.

**Comment by schovi** [comment]
Thanks for explanation. But it is pretty confusing, damn you javascript!

**Comment by leebyron** [comment]
Closed in 59028b5e8552339acbdaab0c3f6547bd131f8f54
